### PR TITLE
TLUNIZIN-658 Implement stand-alone CCM suitable for Performance Testing

### PR DESCRIPTION
**Pull request by dovek**
Due: 11/20
@dlhaines 
@pushyamig

### CODE REVIEW

File: src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java [hunk 1]
```diff
@@ -325,7 +313,7 @@ public void processLti(HttpServletRequest request,
 			M_log.debug("ltiSecret: " + ltiSecret);
 			M_log.debug("ltiUrl: " + ltiUrl);
 		}
-		else {
+		else {
```
Comment (dlhaines) [inline src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java]: White space :-(

File: src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java [hunk 2]
```diff
@@ -408,15 +396,21 @@ private void esbRestApiCall(HttpServletRequest request,
...
-
 	private void apiConnectionLogic(HttpServletRequest request, HttpServletResponse response)
 			throws IOException {
 		PrintWriter out = response.getWriter();
-		if(request.getPathInfo().equalsIgnoreCase(MPATHWAYS_PATH_INFO)){
-			mpathwaysCall(request, response, out);
+		String isStubTesting = appExtPropertiesFile.getProperty(SectionUtilityToolFilter.PROPERTY_TEST_STUB);
+		M_log.debug("isStubTesting: " + isStubTesting);
+		if(Boolean.valueOf(isStubTesting)){
+			Utils.openFile(request, response, out);
```
Comment (dlhaines) [inline src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java]: How about returning right here and then needing the else cause?

File: src/main/java/edu/umich/ctools/sectionsUtilityTool/Utils.java [hunk 3]
```diff
@@ -22,4 +31,129 @@ public static Properties loadProperties(String path){
...
+		if(queryString!=null) {
+			url=pathInfo+"?"+queryString;
+		}else {
+			url=pathInfo;
+		}
+		
+		File testFile = null;
+		FileReader fr = null;
+
+		if (url.matches(SectionsUtilityToolServlet.appExtPropertiesFile.getProperty(CANVAS_API_TERMS))){
```
Comment (dlhaines) [inline src/main/java/edu/umich/ctools/sectionsUtilityTool/Utils.java]: Way too much repeated code.  A couple ways of dealing with that are:
- refactor the try / catch into a method
- Put the pathInfo and associated resource file name into an array and just loop through the array, processing when there is a match.

File: src/main/java/edu/umich/ctools/sectionsUtilityTool/Utils.java [hunk 4]
```diff
@@ -22,4 +31,129 @@ public static Properties loadProperties(String path){
...
+			try{
+				M_log.info("Courses call stub");
```
Comment (dlhaines) [inline src/main/java/edu/umich/ctools/sectionsUtilityTool/Utils.java]: Identical log messages from different locations aren't very helpful.

File: src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java [hunk 5]
```diff
@@ -408,15 +396,21 @@ private void esbRestApiCall(HttpServletRequest request,
...
 
-
 	private void apiConnectionLogic(HttpServletRequest request, HttpServletResponse response)
 			throws IOException {
 		PrintWriter out = response.getWriter();
-		if(request.getPathInfo().equalsIgnoreCase(MPATHWAYS_PATH_INFO)){
-			mpathwaysCall(request, response, out);
+		String isStubTesting = appExtPropertiesFile.getProperty(SectionUtilityToolFilter.PROPERTY_TEST_STUB);
+		M_log.debug("isStubTesting: " + isStubTesting);
+		if(Boolean.valueOf(isStubTesting)){
```
Comment (dlhaines) [inline src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java]: This is better done by using an interface with two implementations, but that can be left for another time.

File: src/main/java/edu/umich/ctools/sectionsUtilityTool/Utils.java [hunk 6]
```diff
@@ -22,4 +31,129 @@ public static Properties loadProperties(String path){
...
+	public static void openFile(HttpServletRequest request,
+			HttpServletResponse response, PrintWriter out) {
+		M_log.debug("Stub method called");
+		String pathInfo = request.getPathInfo();
+		M_log.debug("Path Info: " + pathInfo);
+		
+		String url;
+		String queryString = request.getQueryString();
+		
+		if(queryString!=null) {
```
Comment (dlhaines) [inline src/main/java/edu/umich/ctools/sectionsUtilityTool/Utils.java]: Could be:
 url=pathInfo
if queryString != null then {
url+="?"+queryString
}

File: src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java [hunk 7]
```diff
@@ -408,15 +396,21 @@ private void esbRestApiCall(HttpServletRequest request,
 	 * using Apache Http client library
 	 */
 
-
 	private void apiConnectionLogic(HttpServletRequest request, HttpServletResponse response)
```
Comment (pushyamig) [inline src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java]: Can you add comment about the reason for stubbing?

File: src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java [hunk 8]
```diff
@@ -408,15 +396,21 @@ private void esbRestApiCall(HttpServletRequest request,
...
 			throws IOException {
 		PrintWriter out = response.getWriter();
-		if(request.getPathInfo().equalsIgnoreCase(MPATHWAYS_PATH_INFO)){
-			mpathwaysCall(request, response, out);
+		String isStubTesting = appExtPropertiesFile.getProperty(SectionUtilityToolFilter.PROPERTY_TEST_STUB);
```
Comment (pushyamig) [inline src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java]: The new property should be updated in the Readme.md

**Pull request merged**